src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -203,12 +207,14 @@ internal StructField(JObject json)
         /// <summary>
         /// Returns JSON object describing this type.
         /// </summary>
-        internal object JsonValue =>
-            new JObject(
-                new JProperty("name", Name),
-                new JProperty("type", DataType.JsonValue),
-                new JProperty("nullable", IsNullable),
-                new JProperty("metadata", Metadata));
+        internal object JsonValue => JsonDocument.Parse(JsonSerializer.Serialize(
```

Comment:
You need to dispose the `JsonDocument` and clone the `RootElement` to detach from the arraypool semantics of `JsonDocument`.

See https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.clone?view=netcore-3.0#System_Text_Json_JsonElement_Clone